common/scala/src/main/scala/whisk/common/Logging.scala

```diff
@@ -122,6 +138,45 @@ class PrintStreamLogging(outputStream: PrintStream = Console.out) extends Loggin
 }
 
 /**
+  * Implementation of logging which also supports tracing using Zipkin.
+  * @param logger <code>Logging<code> instance which can do basic logging
+  */
+class ZipkinLogging (logger: Logging) extends Logging {
+
+    def emit(loglevel: LogLevel, id: TransactionId, from: AnyRef, message: String) = {
+        logger.emit(loglevel, id, from, message)
+    }
+
+    override def emit(loglevel: LogLevel, id: TransactionId, from: AnyRef, message: String, logMarker: LogMarker) = {
+
+        //log message as usual
+        emit(loglevel, id, from, TransactionId.createMessageWithMarker(message, logMarker))
+
+        //tracing support
+        logMarker.token.state match {
+
+            case LoggingMarkers.start => {
+                TraceUtil.startTrace(logMarker.token.getServiceName(), id)
+            }
+
+            case LoggingMarkers.finish => {
+                var req = TraceUtil.getTracedRequestForTrasactionId(id)
+                if (req != null)
+                    TraceUtil.finish(id)
+            }
+
+            case LoggingMarkers.error => {
+                var req = TraceUtil.getTracedRequestForTrasactionId(id)
+                if (req != null)
```

Comment:
usually an "error" tag (binary annotation) is added to the span on error. This color-codes it red in the UI